integration-tests/bats/remotes.bats

```diff
@@ -429,6 +429,82 @@ SQL
     [[ "$output" =~ "remotes/test-remote/poop" ]] || false
 }
 
+@test "remotes: dolt fetch check for printing many newlines" {
+    # create_main_remote_branch
+    dolt remote add origin http://localhost:50051/test-org/test-repo
+    dolt sql -q 'create table test (id int primary key);'
+    dolt add .
+    dolt commit -m 'create test table.'
+    dolt push origin main:main
+
+    # create_two_more_remote_branches
+    dolt checkout -b branch1
+    dolt sql -q 'insert into test (id) values (1), (2), (3);'
+    dolt add .
+    dolt commit -m 'add some values to branch 1.'
+    dolt push --set-upstream origin branch1
+
+    dolt checkout -b branch2
+    dolt sql -q 'insert into test (id) values (4), (5), (6);'
+    dolt add .
+    dolt commit -m 'add some values to branch 2.'
+    dolt push --set-upstream origin branch2
+
+    # create first clone
+    cd dolt-repo-clones
+    dolt clone http://localhost:50051/test-org/test-repo
+    cd test-repo
+    dolt status
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch main" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+
+    cd ../..
+    cd "dolt-repo-clones"
+    dolt clone http://localhost:50051/test-org/test-repo test-repo2
+    cd test-repo2
+    dolt status
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch main" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+
+    # CHANGE 1: add more data to branch1
+    dolt checkout -b branch1 remotes/origin/branch1
+    dolt sql -q 'insert into test (id) values (100), (101), (102);'
+    dolt add .
+    dolt commit -m 'add more values to branch 1.'
+    dolt push --set-upstream origin branch1
+
+    # CHANGE 2: add more data to branch1
+    dolt checkout -b branch2 remotes/origin/branch2
+    dolt sql -q 'insert into test (id) values (103), (104), (105);'
+    dolt add .
+    dolt commit -m 'add more values to branch 2.'
+    dolt push --set-upstream origin branch2
+
+    # CHANGE 3: create_remote_branch "branch-three"
+    dolt checkout -b branch3
+    dolt sql -q 'insert into test (id) values (7), (8), (9);'
+    dolt add .
+    dolt commit -m 'add some values to branch 3.'
+    dolt push --set-upstream origin branch3
+
+    # CHANGE 4: create_remote_branch "branch-four"
+    dolt checkout -b branch4
+    dolt sql -q 'insert into test (id) values (10), (11), (12);'
+    dolt add .
+    dolt commit -m 'add some values to  branch 4.'
+    dolt push --set-upstream origin branch4
+
+    cd ..
+    cd test-repo
+    run dolt fetch
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 4 ]
```

Comment:
Does this test what you want it to?

The bug is dolt fetch producing many blank lines when it should not. I think you need another test case for that behavior (where the number of lines of output is one, and it's just a newline). Here's you're testing that there are 4 lines of output. If those are legitimate, you should make assertions about their content.

Reply:
added uptodate fetch output check